core/tee/tee_svc_storage.c

```diff
@@ -924,6 +941,26 @@ static TEE_Result tee_svc_storage_set_enum(char *d_name, struct tee_obj *o)
 
 }
 
+static TEE_Result tee_svc_storage_verify_enum(struct tee_ta_session *sess,
+					      struct tee_obj *o, char *d_name)
+{
+	TEE_Result res;
+
+	/* allocate obj_id and set object */
+	res = tee_svc_storage_set_enum(o, d_name);
+	if (res != TEE_SUCCESS)
+		goto exit;
+	res = tee_obj_verify(sess, o);
+	if (res != TEE_SUCCESS)
+		goto exit;
+	/* free obj_id */
+	free(o->pobj->obj_id);
```

Comment:
Move this below `exit:` and set `o->pobj->obj_id` to NULL also and we stop fiddling with `o->pobj->obj_id` at other places.